crates/ra_assists/src/auto_import.rs

```diff
@@ -334,7 +334,7 @@ fn best_action_for_target<'b, 'a: 'b>(
         .filter_map(ast::UseItem::use_tree)
         .map(|u| walk_use_tree_for_best_action(&mut storage, None, u, target))
         .fold(None, |best, a| {
-            best.and_then(|best| Some(*ImportAction::better(&best, &a))).or(Some(a))
+            best.and_then(|best| Some(*ImportAction::better(&best, &a))).or_else(|| Some(a))
```

Comment:
I feel conflicted about this, it seems like closure creates noise for little benefit here. But I guess it's ok?

Reply:
Yeah, I think the fact that clippy interprets these constructors as "function calls" is pretty unfortunate, It looks like a bug (https://github.com/rust-lang/rust-clippy/issues/1609), but for now I figured that it'd be better to make the changes just so we can get all the lint errors out of the way.